### PR TITLE
Update docs cell description vs. cell kind justification

### LINKED ISSUE #1236: Fix tone of docs
There is an attempt to justify whey recipes require callbacks for both cell description and cell type in a blue `note!` box here:

https://arbor.readthedocs.io/en/latest/concepts/recipe.html#are-recipes-always-neccessary

I think that this is a bit problematic for two reasons:
* the tone is not right for serious documentation.
* it does not really describe the performance rationale.

I will reword it.

### PR DESCRIPTION
**Pull request by bcumming**
Fixes #1236

### CODE REVIEW

File: doc/concepts/recipe.rst [hunk 1]
```diff
@@ -113,16 +113,16 @@ The steps of building a simulation from a recipe are:
...
-    | Recipe: it's a foobar.
-    | Simulator: Okay.
-    | Cell group implementations: which one of you lot deals with foobars?
-    | Foobar_GPUFTW_lolz: That'd be me, if we've got GPU enabled.
-    | Simulator: Okay it's up to you then to deal with this s3cr1ts4uc3 object.
+
+    It might seem like overkill to have a separate call that returns the cell kind, when one could determine
+    the kind by requesting the cell description, then querying the kind of the result.
+
+    However, in some parts of model building only require the cell kind, and not
```
Comment (halfflat) [inline doc/concepts/recipe.rst:120]: I'd say "Some phases in the construction of a simulation, however, require only the cell kind, and not..."

File: doc/concepts/recipe.rst [hunk 2]
```diff
@@ -113,16 +113,16 @@ The steps of building a simulation from a recipe are:
...
+    the full cell description. Cell descriptions can be quite expensive to
+    assemble, for example, building a Purkinje cell model with complex geometry and
```
Comment (halfflat) [inline doc/concepts/recipe.rst:122]: I'd rephrase slightly just for grammar: "assemble; for example, a Purkinje cell model can have very complex geometry, a rich collection of ion channels, and thousands of synapses."

File: doc/concepts/recipe.rst [hunk 3]
```diff
@@ -113,16 +113,16 @@ The steps of building a simulation from a recipe are:
...
+    rich ion channel and thousands of synapses.
+    To avoid the overhead of constructing a full cell model when only the cell kind
+    is required, the recipe provides a separate callback that returns only the cell kind.
```
Comment (halfflat) [inline doc/concepts/recipe.rst:126]: I don't think we need this last sentence, given the context.

**Review by halfflat — CHANGES_REQUESTED** [review]
Totally approve of the intent and content; comments are just there to tighten the text and/or address some minor grammatical issues.

File: doc/concepts/recipe.rst (hunk 2 above)
Comment (bcumming) [reply doc/concepts/recipe.rst:122]: You know I like a semicolon.

**Review by halfflat — APPROVED** [review]
